### PR TITLE
Destroy environment workflow

### PR DESCRIPTION
**Pull request by gurevichdmitry**
### Summary of your changes

1. Added a new Destroy environment GitHub workflow: A new GitHub workflow was created to handle the destruction of resources. It requires three inputs: `EC_API_KEY`, `prefix`, and `ignore-prefix`. These inputs are essential for identifying the resources to be deleted and ensuring that the correct ones are removed.

2. Extended script to delete CloudFormation stacks: The existing cleanup script was enhanced to include the deletion of CloudFormation stacks. The script now retrieves a list of CloudFormation stacks that match the provided `ENV_PREFIX` and `IGNORE_PREFIX`, if specified. It then iterates through the list and deletes each CloudFormation stack using the `aws cloudformation delete-stack` command. Additionally, the script now waits until the deletion of each stack is completed before moving on to the next one, ensuring a clean and thorough removal of resources.

3. Updated documentation 

[Workflow run example](https://github.com/gurevichdmitry/cloudbeat/actions/runs/5608765275)

### Checklist
- [ ] I have added tests that prove my fix is effective or that my feature works
- [ ] I have added the necessary README/documentation (if appropriate)

### CODE REVIEW

File: .github/workflows/destroy-environment.yml [hunk 1]
```diff
@@ -0,0 +1,54 @@
+name: Destroy Environment
+run-name: Destroying environments starting with ${{ github.event.inputs.prefix }} by @${{ github.actor }}
```
Comment (oren-zohar) [inline .github/workflows/destroy-environment.yml:2]: maybe something shorter?
```suggestion
run-name: Destroying ${{ github.event.inputs.prefix }}* by @${{ github.actor }}
```

File: .github/workflows/destroy-environment.yml [hunk 2]
```diff
@@ -0,0 +1,54 @@
...
+  # Ability to execute on demand
+  workflow_dispatch:
+    inputs:
+      ec-api-key:
+        type: string
+        description: "Elastic Cloud API KEY"
+        required: true
+      prefix:
+        type: string
+        description: "delete all cloud environments that start with {{prefix}}"
```
Comment (oren-zohar) [inline .github/workflows/destroy-environment.yml:14]: can we use `prefix` here?
```suggestion
        description: "Delete all environments starting with `prefix`"
```

File: .github/workflows/destroy-environment.yml [hunk 3]
```diff
@@ -0,0 +1,54 @@
...
+        required: true
+      ignore-prefix:
+        type: string
+        description: "delete all cloud environments that do not start with {{ignore-prefix}}"
```
Comment (oren-zohar) [inline .github/workflows/destroy-environment.yml:18]:
```suggestion
        description: "Ignore all environments starting with `ignore-prefix`"
```

File: dev-docs/Cloud-Env-Testing.md [hunk 4]
```diff
@@ -121,3 +121,36 @@ Before running the script, ensure that:
 
 **Note**: The script will ask for confirmation before deleting each environment, unless you set the `interactive` flag
 to `false`.
+
+### Delete Environment Workflow
```
Comment (oren-zohar) [inline dev-docs/Cloud-Env-Testing.md:125]: I think this part should go first (before the script info)

File: dev-docs/Cloud-Env-Testing.md [hunk 5]
```diff
@@ -121,3 +121,36 @@ Before running the script, ensure that:
...
+2. Click the `Run workflow` button.
+
+   ![Run Workflow](https://github.com/gurevichdmitry/cloudbeat/assets/99176494/8965311c-eeac-492f-a564-a57c46854a3a)
+
+3. Complete the required input fields:
+
+    - `ec-api-key` (required): Use the [Production Elastic Cloud](https://cloud.elastic.co/home) API KEY.
+    - `prefix` (required): The prefix used to identify the environments to be deleted.
+
+   ![Enter Inputs](https://github.com/gurevichdmitry/cloudbeat/assets/99176494/dde955f1-21fe-4cb1-a7bc-674d8969b001)
```
Comment (oren-zohar) [inline dev-docs/Cloud-Env-Testing.md:146]: can we resize the image size? https://gist.github.com/uupaa/f77d2bcf4dc7a294d109

File: dev-docs/Cloud-Env-Testing.md [hunk 6]
```diff
@@ -121,3 +121,36 @@ Before running the script, ensure that:
...
+
+4. Optionally, modify other input values if required:
+
+    - `ignore-prefix` (optional): The prefix used to identify environments that should be excluded from deletion.
+
+   ![Optional Inputs](https://github.com/gurevichdmitry/cloudbeat/assets/99176494/5b4f732f-16b9-4b97-8c1c-6955e66d387b)
```
Comment (oren-zohar) [inline dev-docs/Cloud-Env-Testing.md:152]: can we resize the image here? I think it's way too big (https://gist.github.com/uupaa/f77d2bcf4dc7a294d109)

**Review by oren-zohar — CHANGES_REQUESTED** [review]

File: dev-docs/Cloud-Env-Testing.md (hunk 4 above)
Comment (gurevichdmitry) [reply dev-docs/Cloud-Env-Testing.md]: Updated

File: dev-docs/Cloud-Env-Testing.md (hunk 5 above)
Comment (gurevichdmitry) [reply dev-docs/Cloud-Env-Testing.md]: Updated

File: dev-docs/Cloud-Env-Testing.md (hunk 6 above)
Comment (gurevichdmitry) [reply dev-docs/Cloud-Env-Testing.md]: updated

File: dev-docs/Cloud-Env-Testing.md (hunk 5 above)
Comment (gurevichdmitry) [reply dev-docs/Cloud-Env-Testing.md]: new images uploaded

File: dev-docs/Cloud-Env-Testing.md (hunk 6 above)
Comment (gurevichdmitry) [reply dev-docs/Cloud-Env-Testing.md]: new image uploaded

File: dev-docs/Cloud-Env-Testing.md [hunk 7]
```diff
@@ -104,6 +104,43 @@ Follow these steps to log in to the created environment:
...
+
+   ![Destroy Environment](https://github.com/gurevichdmitry/cloudbeat/assets/99176494/505d6553-7780-4450-83e9-3617f64442ad)
+
+2. Click the `Run workflow` button.
+
+   ![Run Workflow](https://github.com/gurevichdmitry/cloudbeat/assets/99176494/8965311c-eeac-492f-a564-a57c46854a3a)
+
+3. Complete the required input fields:
+
+    - `ec-api-key` (required): Use the [Production Elastic Cloud](https://cloud.elastic.co/home) API KEY.
```
Comment (oren-zohar) [inline dev-docs/Cloud-Env-Testing.md:127]:
```suggestion
    - `ec-api-key` (required): Use the [Production Elastic Cloud](https://cloud.elastic.co/home) API key.
```

File: .github/workflows/destroy-environment.yml [hunk 8]
```diff
@@ -0,0 +1,54 @@
+name: Destroy Environment
+run-name: Destroying ${{ github.event.inputs.prefix }}* by @${{ github.actor }}
+
+on:
+  # Ability to execute on demand
+  workflow_dispatch:
+    inputs:
+      ec-api-key:
+        type: string
+        description: "Elastic Cloud API KEY"
```
Comment (oren-zohar) [inline .github/workflows/destroy-environment.yml:10]:
```suggestion
        description: "Elastic Cloud API key"
```

**Review by oren-zohar — APPROVED** [review]

**Review by oren-zohar — APPROVED** [review]

**Pull request merged**